File: semestr_3/aisd/study/3_insertion_sort.cpp
#include <iostream>
#include <vector>
using namespace std;

void wypisz(int *tab, int n){
    for(int i =0; i < n; i++)
        cout << tab[i] << ", ";
    cout << endl;
}

void insertion_sort(int *tab, int n){
    int i, j, value;
    for(i = 1; i < n; i++){
        value = tab[i];
        j = i;
        while(j>0 && tab[j-1] > value){
            tab[j] = tab[j-1];
            j -= 1;
        }
        tab[j] = value;
    }
}

int main(){
    int arr[]{3,4,5,2,1};
    int arrSize = sizeof(arr)/ sizeof(arr[0]);

    wypisz(arr, arrSize);
    insertion_sort(arr,arrSize);
    wypisz(arr, arrSize);

    return 0;
}

File: semestr_3/aisd/study/3_insertion_sort.py
def insertion_sort(array):
    for i in range(1, len(array)):
        # Wartosc przy granicy
        value = array[i]
        j = i
        while j > 0 and array[j-1] > value:
            array[j] = array[j-1]
            j -= 1
        array[j] = value

arr = [7,4,8,3,2,1,5,6]
print(arr)
insertion_sort(arr)
print(arr)
